### DISCUSSION TITLE
React Query stops when component is not visible

### DISCUSSION THREAD

**Discussion by rasmusfiskerbang** [TanStack/query#1564] [Q&A] [answered]
Hi,

I have a modal with a button and an attached callback that closes the modal and then calls `refetch`. 
I can see that my `queryFn` is in fact working, however, `onSuccess` never gets called. I guess that's because the modal is closed, and as a consequence React Query does not think the data is in use, but in my situation I'm adding it to a Redux store for later user manipulation, using onSuccess.

- Why does the query not continue?
- Have I missed the purpose of the library, or have I overlooked some setting?

Thanks

**Comment by TkDodo** [comment, marked as answer]
It could be that `onSuccess` is not called because the subscriber has already unmounted. Can you create a reproduction in codesandbox?

if that’s the case, you could try `queryClient.fetchQuery` rather than `refetch`

**Reply by rasmusfiskerbang** [reply]
That was the case.

**Reply by TkDodo** [reply]
cool. Did `queryClient.fetchQuery` work for you?

**Reply by TkDodo** [reply]
The difference might be that `refetch` works on the `observer` created by `useQuery`, which does not exist anymore when your component is unmounted and thus the subscription does not exist anymore, while `queryClient` is static and the `fetchQuery / refetchQueries` methods will just execute the fetch if necessary and then notify all living subscribers to update - so that should work
